Workflow to auto-lock old closed issues and PRs

hamishwillee opened the pull request:
This adds a workflow to automatically lock issues and PRs that have been closed for more than a year. 

This should reduce the instance of "necrobumping" - adding follow-on discussions to issues that are long closed and often not relevant. Even if used in "good faith", this is not a good idea because the old issue was addressed, and the detail on a year old issue is likely to be out of date or incorrect.

The implementation is directly copied from https://github.com/dessant/lock-threads#examples and uses all defaults:
- Locks issues after 365 days of a PR or issue being inactive.

NOTE: I personally think this is a good idea, and there was some support on slack. I have not yet "tested" this.

hamishwillee committed:
Workflow to auto-lock old closed issues and PRs

wbamberg commented:
I think this is a good idea, but I'm not particularly qualified to review it and can't merge it (I think), so removing my request for review. @fiji-flo , @Rumyra , wdyt?

Rumyra commented:
I tell you what would be good - if this was added to [mdn/workflows](https://github.com/mdn/workflows) as a shared workflow across all repos @hamishwillee - pinging @schalkneethling so he's aware 👍

schalkneethling commented:
> I tell you what would be good - if this was added to [mdn/workflows](https://github.com/mdn/workflows) as a shared workflow across all repos @hamishwillee - pinging @schalkneethling so he's aware 👍

Yes, please! 😸

hamishwillee commented:
Cool. I've copied this across to https://github.com/mdn/workflows/pull/38 - do I need to do anything else?

schalkneethling commented:
> Cool. I've copied this across to [mdn/workflows#38](https://github.com/mdn/workflows/pull/38) - do I need to do anything else? It would be really good if this could be tested in a safe place :-)

Nope, we can discuss it further in the related pull request.

> It would be really good if this could be tested in a safe place :-)

Indeed.

hamishwillee committed:
lock closed copy idle chron

hamishwillee commented on .github/workflows/lock-closed.yml:
```diff
@@ -0,0 +1,8 @@
+name: 'Lock old issues and pull requests'
+on:
+  schedule:
+    - cron: '49 11,23 * * *'
```
@schalkneethling Updated, but I **really** don't know what I am doing here. In honesty, I'd hoped that if I found a workflow that did this, someone with more expertise would roll it out. 

Anyway, I copied the same schedule as the [idle workflow](https://github.com/mdn/content/blob/main/.github/workflows/idle-issues.yml). So this should run whenever that does, with hard coded values for locking to happen one year after closure.

schalkneethling commented:
Hey there @hamishwillee! Apologies for the silence here. I am making some updates to the workflows in `mdn/workflows` so they only run on the, in most cases, origin repo and not on all forks. I will also test this one out on one of our smaller repos and get back to you here.

Commit:
Merge branch 'main' into lock_closed_issues_prs

hamishwillee commented:
Thanks! Let me know if you need me to do anything here. Github actions is not one of my strengths so I am hoping you don't need me :-0

schalkneethling commented:
> Thanks! Let me know if you need me to do anything here. Github actions is not one of my strengths so I am hoping you don't need me :-0

Will do, thank you @hamishwillee

Commit:
chore: add target-repo

Adds `target-repo` flag to ensure this only runs on `mdn/content` and not on all forks as well.

Commit:
Merge branch 'main' into lock_closed_issues_prs

schalkneethling reviewed (approved):
Thanks @hamishwillee, let’s take this for a spin :)

hamishwillee commented:
Thanks @schalkneethling . I was kind of expecting a rush of auto-locks but no notifications yet.

schalkneethling commented:
> Thanks @schalkneethling . I was kind of expecting a rush of auto-locks but no notifications yet.

Yup, it failed :) I have logged the bug and will take care of it today.
